torch/_functorch/_aot_autograd/runtime_wrappers.py

```diff
@@ -248,6 +255,106 @@ def _should_disable_saved_tensors_hooks():
     return False
 
 
+def _schema_allows_aliasing(func) -> bool:
+    schema = func._schema
+    # View ops have non-write aliases declared in arguments
+    if schema._is_view_op():
+        return True
+    # Handles cases like mkldnn::_convolution_pointwise_.binary
+    # where the schema is Tensor(a!) other -> Tensor(a!) Y
+    for ret in schema.returns:
+        if ret.alias_info is not None:
+            return True
+    return False
+
+
+def _check_custom_op_aliasing(name, args, kwargs, result):
+    """
+    Check if custom op outputs alias inputs or other outputs.
+    If config.check_custom_op_mode is True, raises RuntimeError.
+    Otherwise, emits a warning.
+    """
+    try:
+        torch._library.utils._c_check_aliasing_constraint(
+            name,
+            args,
+            kwargs,
+            result,
+        )
+    except RuntimeError as e:
+        if config.check_custom_op_mode:
```

Comment:
It looks to me like the main purpose of this config is to decide if we should warn vs error.

Doesn't that mean that the new TorchDispatchMode is effectively on-by-default (via warning) once this PR lands?The risk to me is that if this mode is sufficiently slow, then we are regressing torch.compile runtime perf / overheads noticeably by default.

cc @zou3519 to confirm, but I think what I would imagine is something like:

(1) we still need another config to decide if we are using the mode **at all**, due to overhead risk (independent of whether the mode should warn vs error)

(2) if we want the mode to run by default, then we need to do more work to measure the overhead, and if it is bad then potentially avoid using a TorchDispatchMode (via moving the checks into inductor codegen).